app/lib/actions/pulp3/orphan_cleanup/delete_orphan_repository_versions.rb

```diff
@@ -7,12 +7,7 @@ def plan(smart_proxy)
         end
 
         def run
-          smart_proxy = SmartProxy.find(input[:smart_proxy_id])
-          if smart_proxy.pulp_mirror?
-            output[:pulp_tasks] = ::Katello::Pulp3::Repository.delete_orphan_repository_versions_for_mirror(smart_proxy)
-          else
-            output[:pulp_tasks] = ::Katello::Pulp3::Repository.delete_orphan_repository_versions(smart_proxy)
-          end
+          output[:pulp_tasks] = ::Katello::Pulp3::SmartProxyRepository.instance_for_type(smart_proxy).delete_orphan_repository_versions
```

Comment:
Don't see where smart_proxy is defined before this variable

Reply:
there's a helper method in pulp3/abstract.rb